lib/OpenQA/WebAPI/Plugin/Gru.pm

```diff
@@ -59,65 +90,74 @@ sub enqueue {
     my ($self, $task) = (shift, shift);
     my $args = shift // [];
     my $options = shift // {};
+    my $jobs = shift // [];
 
-    $self->schema->resultset('GruTasks')->create(
+    $args = [$args] if ref $args eq 'HASH';
+
+    my $delay = $options->{run_at} && $options->{run_at} > now() ? $options->{run_at} - now() : 0;
+
+    my $gru = $self->schema->resultset('GruTasks')->create(
         {
             taskname => $task,
-            args     => $args,
             priority => $options->{priority} // 0,
-            run_at   => $options->{run_at} // now()});
+            args     => $args,
+            run_at   => $options->{run_at} // now(),
+            jobs     => $jobs,
+        });
+
+    $self->app->minion->enqueue(
+        $task => $args => {priority => $options->{priority} // 0, delay => $delay, notes => {gru_id => $gru->id}});
 }
 
 package OpenQA::WebAPI::Plugin::Gru::Command::gru;
 use Mojo::Base 'Mojolicious::Command';
+use Mojo::Pg;
+use Data::Dumper;
 
 has usage       => "usage: $0 gru [-o]\n";
 has description => 'Run a gru to process jobs - give -o to exit _o_nce everything is done';
 
+sub delete_gru {
+    my ($self, $id) = @_;
+    my $gru = $self->app->db->resultset('GruTasks')->find($id);
+    $gru->delete() if $gru;
+}
+
 sub cmd_list {
     my ($self) = @_;
-
-    my $tasks = $self->app->schema->resultset('GruTasks');
-    while (my $task = $tasks->next) {
-        use Data::Dumper;
-        print $task->taskname . " " . Dumper($task->args) . "\n";
+    my $tasks = $self->app->minion->backend->list_jobs();
```

Comment:
The `list_jobs` method technically requires `$offset` and `$limit` arguments. It appears you've stumbled over an unintended feature where both end up as `undef`, which `DBD::Pg` turns into `LIMIT NULL OFFSET NULL`, and which PostgreSQL just so happens to understand. Not sure if we can support this officially as a feature in Minion.

Reply:
not sure even if job list is actually used - if we end up exposing the minion admin interface i guess we can drop it, @coolo ?

Reply:
There's also the `minion job` command, which has list functions for everything on the command line.

Reply:
http://mojolicious.org/perldoc/Minion/Command/minion/job#SYNOPSIS

Reply:
@kraih thanks for the pointers, should be handled now in https://github.com/os-autoinst/openQA/pull/1620/commits/ceb8e55cb003f127ae72c1a1844044155653832a